test/functional/wallet_labels.py

```diff
@@ -25,6 +25,42 @@ def set_test_params(self):
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
+    def invalid_label_name_test(self):
+        node = self.nodes[0]
+        address = node.getnewaddress()
+        pubkey = node.getaddressinfo(address)['pubkey']
+        rpc_calls = [
+            [node.getnewaddress],
+            [node.setlabel, address],
+            [node.getaddressesbylabel],
+            [node.importpubkey, pubkey],
+            [node.addmultisigaddress, 1, [pubkey]],
+            [node.getreceivedbylabel],
+        ]
+        for rpc_call in rpc_calls:
+            assert_raises_rpc_error(-11, "Invalid label name", *rpc_call, "*")
+
+        r = None
+        if self.options.descriptors:
+            r = node.importdescriptors([{
+                'desc': 'pkh(' + address + ')',
+                'label': '*',
+                'timestamp': 'now',
+            }])
+        else:
+            privkey = node.dumpprivkey(address)
+            assert_raises_rpc_error(-11, "Invalid label name", node.importprivkey, privkey, "*")
```

Comment:
nit: if you move the entire if/else loop to before the execution of the rpc calls, you could replace this with `rpc_calls.append(...)`